src/controllers/components/model.js

```diff
@@ -1,12 +1,30 @@
-import { searchModel } from '../../lib/api-client';
+import { searchModel, deleteModel, getAllModels } from '../../lib/api-client';
 import { Dispatcher } from 'consus-core/flux';
 
 export default class ModelController {
     static getModel(address) {
         return searchModel(address).then( model => {
-            Dispatcher.handleAction("MODEL_FOUND", model);
+            Dispatcher.handleAction('MODEL_FOUND', model);
         }).catch( () => {
-            Dispatcher.handleAction("NO_MODEL_FOUND");
+            Dispatcher.handleAction('NO_MODEL_FOUND');
+        });
+    }
+
+    static deleteModel(address){
+        return deleteModel(address).then(data => {
+            Dispatcher.handleAction('MODEL_DELETED', data.deletedModel);
+        }).catch(error => {
+            Dispatcher.handleAction('ERROR', {
+                error: error.message
+            });
+        }).then(() => {
+            return getAllModels().then(models => {
+                Dispatcher.handleAction("MODELS_RECEIVED", models);
```

Comment:
Can't we update the stores using `MODEL_DELETED`? The extra step here seems unnecessary.

Also, should the items be updated? Are the items of a deleted model also deleted?